src/commands/exec.ts

```diff
@@ -58,20 +70,17 @@ export default class Exec extends BaseCommand {
   public static readonly StderrStream = 2;
   public static readonly StatusStream = 3;
 
-  async exec(uri: string): Promise<void> {
-    const { args, flags } = await this.parse(Exec);
-
+  async exec(uri: string, flags: ExecFlags): Promise<void> {
     const ws = await this.getWebSocket(uri);
 
     await new Promise((resolve, reject) => {
       const duplex = createWebSocketStream(ws, { encoding: 'utf-8' });
       duplex.pipe(this.getOutputTransform());
 
       if (flags.stdin) {
-        if (flags.tty && !process.stdin.isTTY) {
-          throw new ArchitectError('stdin does not support tty');
+        if (flags.tty) {
+          process.stdin.setRawMode(true);
```

Comment:
This was an existing bug that's now fixed, `process.stdin.setRawMode` is only available if stdin is a TTY: https://nodejs.org/api/tty.html#readstreamsetrawmodemode (it's only part of the `tty.ReadStream` class)

Reply:
Nice find. Could you add this as a comment in the code?